Implement a more complete Java Client

seijikun opened the pull request:
During your course for the FHWS, a couple of us thought it would be cool if there was a possibility to get a listing of currently open games, since this would speed up the testing process quite a bit.
(Getting a list of open games via telnet, copy pasting a free gameId into the client code, noticing that someone else stole it just now, 😠, doing it again...)

This Client adds support for:
- Using it without login (certain commands)
- Retrieving listings (such as ratings and games)
- Requesting a rerate of the authenticated player

A small usage example:
```Java
//connect to the server
Client2 client = new Client2("imcs.svcs.cs.pdx.edu", 3589);

//get a list of ratings (since we want to see who we're going to be dealing with!)
Map<String, Integer> ratings = client.getRatingsList();

//get a free game offer where we can be player white (since we only support white!)
Optional<String> freeGameId = client.getGameList().stream()
		.filter(g -> !g.isRunning && g.reservedPlayer == 'B')
		.map(g -> g.gameId)
		.findFirst();

//log into the server with our account and password
client.login("lazycat", "obviouslyNotMyPassword!");

if(freeGameId.isPresent()) { //there is a matching game, accept the offer
	client.accept(freeGameId.get());
} else { //there is no game matching our rules, create one where we are white
	client.offerGameAndWait('W');
}

// ... play some super intensive game ...

//request a rerate so that noone can see how bad our opponent destoryed us just now
client.rerate();

//close connection
client.close();
```

I wrote this client anyway, so I thought you might appreciate the contribution.
I wasn't sure if you want a client that needs multiple files, so I didn't change the existing one.

Thank you very much for doing this course, despite your health condition!
As far as I can see, everyone of us is enjoying it very much.

BartMassey commented:
Wow, nice piece of work! Please restructure the pull request so that it replaces my code instead of adding to it. We can then forget my code ever existed. :-)

Thanks for contributing this.

seijikun commented:
Okay, I replaced the old client.
Since there are no two clients anymore, I refactored it's name to `Client` and fixed two bugs.
I also generated the JavaDocs.

BartMassey commented:
Thanks again. This is a great contribution that will help people a lot in the future.
